### ISSUE TITLE
Import GLTF curves physical skeleton not exact place

### ISSUE THREAD

**Issue by Magical10240** [godotengine/godot#39757]
**Godot version:**
v3.2.1stable/v3.2.2rc2

**OS/device including version:**
Debian 10.4

**Issue description:**
Blender 2.83 export import format GLTF/DAE problem miscellaneous left or right
![2020-06-22_17-29](https://user-images.githubusercontent.com/17591539/85301762-e6581f00-b4b0-11ea-96c5-608777ee7302.png)

already tried godot 3.2.1stable and 3.2.2rc2 anyway it's useless who hasn't fixed physical bones

blender 2.83 the original modeling + skeletons
![image](https://user-images.githubusercontent.com/17591539/85302033-4fd82d80-b4b1-11ea-9115-bb4837ef7737.png)

**Minimal reproduction project:**
[_cloth_phys.blend.zip](https://github.com/godotengine/godot/files/4813944/_cloth_phys.blend.zip) only blender 2.83

**Comment by MadaraZ** [comment]
test model took from youtube tutorial https://www.youtube.com/watch?v=mdyRO8_q7lI

**Comment by mohammedzero43** [comment]
This is an issue with the mesh it self

The mesh transformations were not applied before exporting 👌


However, it's true that GLTF importer in godot has an issue with identifying bones orientation when creating physical bones

**Comment by fire** [comment]
[_cloth_phys.blend.zip](https://github.com/godotengine/godot/files/4824109/_cloth_phys.blend.zip)

I can create a fixed version of this.

Using the gltf2 on 3.2 and the blender option set to "fortune".

![Screen Shot 2020-06-24 at 1 38 40 AM](https://user-images.githubusercontent.com/32321/85523151-75515e00-b5bb-11ea-8e3c-22870f6f8f17.png)

**Comment by MadaraZ** [comment]
this i made the result useless nothing has changed bones

![2020-06-24_17-15](https://user-images.githubusercontent.com/17591539/85573371-80999d80-b63e-11ea-9cac-6c781bd0cbec.png)

**Comment by fire** [comment]
I'm not able to dig deeply into find the cause today. However, I can post the scene for my variation if it can help.

![editor_screenshot_2020-06-24T080517-0700](https://user-images.githubusercontent.com/32321/85581644-9df44a80-b5f1-11ea-813f-6754372f6f8c.png)

I can confirm it fails for rc2.

**Comment by fire** [comment]
Are you able to retest on latest stable?

**Comment by Blackiris** [comment]
Still reproducible in Godot 3.2.4 beta 4

**Comment by Blackiris** [comment]
From what I understand, this issue could be from 2 different causes: error from the Create physical skeleton code, or error from the GLTF (parsing, or export from Blender).
Given that GLTF animation seems to work quite well for animation, I have looked at the first cause.

In skeleton_editor_plugin.cpp, the code is assuming the bone direction is pointing to -Z, hence the physical bone body offset to put it in the middle:

`body_transform.origin = Vector3(0, 0, -half_height);`

But, when using GLTF model, it could point to completely another direction
By setting the body offset using the bone rest vector instead, it partially fixes the issue:

```
Vector3 bone_rest_origin = skeleton->get_bone_rest(bone_child_id).origin;
body_transform.origin = bone_rest_origin * 0.5;
```

![bone_origin_ok_dir_ko](https://user-images.githubusercontent.com/440909/102398861-1d326c00-3fe0-11eb-9882-6d7ac56e65ca.png)

However, the physical bone rotation is not set along the bone, and for this, I had to use:

`body_transform = body_transform.looking_at(Vector3(0, 0, 0), Vector3(0, 0, 1));`

![bone_origin_ok_dir_ok](https://user-images.githubusercontent.com/440909/102399404-e3ae3080-3fe0-11eb-9c27-b9d360ebbe14.png)

It is my first look at Godot's code so I don't know if it is the proper way to do the rotation...

**Comment by Blackiris** [comment]
It was not a good idea to rotate the physical bones since it creates issues with the physic engine. Instead, rotation the collision shape is much safer.

**Comment by Blackiris** [comment]
I have just created a PR for the Godot 4.0 version #48310.
Please find here a reproduction project for Godot 4.0 containing 3 models, using DAE and GLTF (total 6 cases):

[testSkeleton-godot4-0.zip](https://github.com/godotengine/godot/files/6401625/testSkeleton-godot4-0.zip)

Just open one of the 6 scene, select the skeleton and click on Create physical skeleton.

**Comment by akien-mga** [comment]
Fixed by #48310.
